classes/Api/Payment/Onboarding.php

```diff
@@ -31,122 +32,35 @@ class Onboarding extends PaymentClient
-            'email_address' => \Configuration::get('PS_PSX_FIREBASE_EMAIL'),
-            'name' => [
-                'given_name' => $psxFormData['business_contact_first_name'],
-                'surname' => $psxFormData['business_contact_last_name'],
-                'prefix' => $psxFormData['business_contact_gender'],
-            ],
-        ];
-
-        return array_filter($nameObj);
-    }
-
-    /**
-     * Generate an array to be used on the Paypal Link
-     *
-     * @param array $psxFormData
-     *
-     * @return array
-     */
-    private function getBusinessDetails($psxFormData)
-    {
-        $nameObj = [
-            'business_address' => array_filter([
-                'city' => $psxFormData['business_address_city'],
-                'country_code' => $psxFormData['business_address_country'],
-                'line1' => $psxFormData['business_address_street'],
-                'postal_code' => $psxFormData['business_address_zip'],
-                'state' => $psxFormData['business_address_state'],
-            ]),
-            'phone_contacts' => [
-                0 => [
-                'phone_number_details' => [
-                        'country_code' => (string) $psxFormData['business_phone_country'],
-                        'national_number' => $psxFormData['business_phone'],
-                    ],
-                    'phone_type' => 'HOME',
-                ],
-            ],
-            'names' => [
-                0 => [
-                    'name' => $psxFormData['shop_name'],
-                    'type' => 'LEGAL',
-                ],
-            ],
-            'category' => $psxFormData['business_category'],
-            'sub_category' => $psxFormData['business_sub_category'],
-            'website_urls' => array_filter([
-                $psxFormData['business_website'],
-            ]),
-            'business_type' => 'INDIVIDUAL',
-            'average_monthly_volume_range' => (new PsxDataMatrice())->getCompanyEmrToAverageMonthlyVolumeRange($psxFormData['business_company_emr']),
-        ];
-
-        return array_filter($nameObj);
-    }
+        //TODO: If the response is 400 with fullPayload retry the call with the minimal payload
 
-    /**
-     * Generate the callback url used by the paypal button
-     *
-     * @return string callback link
-     */
-    private function getCallBackUrl()
-    {
-        return $this->link->getAdminLink('AdminPaypalOnboardingPrestashopCheckout');
-    }
+        if (false === $response['status']) {
+            return $response;
+        }
 
-    /**
-     * Get the iso code of the default currency for the shop
-     *
-     * @return string iso code
-     */
-    private function getCurrencyIsoCode()
-    {
-        $currency = \Currency::getCurrency((int) \Configuration::get('PS_CURRENCY_DEFAULT'));
+        if (false === isset($response['body']['links']['1']['href'])) {
+            return $response['status'] = false;
+            return $response;
```

Comment:
$response is a homemade array

Reply:
Yeah this homemade array is the improvement brought by this PR, but that's not the point here. If you look at the controller requesting the onboardling link, then at the VueJs state, I'm not sure they are expecting the whole array as a response.

In the vueJs component, the following check is done: 

```js
getOnboardingLink() {
        if (this.paypalOnboardingLink !== false) {
          this.$refs.paypalButton.click();
          return;
        }
```
In case of error with the PayPal API, the whole response array will be returned from the ajax request, and the status is checked nowhere. That's why I was more expecting a `return false;` here.